js/integration-tests/solidity-tests/test/testContext.ts

```diff
@@ -104,17 +106,36 @@ export class TestContext {
       throw new Error(`No matching test contract found for ${contractName}`);
     }
 
-    const suiteResults = await runAllSolidityTests(
-      this.edrContext,
-      chainType,
-      this.artifacts,
-      testContracts,
-      this.tracingConfig,
-      {
-        ...this.defaultConfig(chainType),
-        ...config,
-      }
-    );
+    let generate_gas_report = config?.generateGasReport ?? false;
+    let suiteResults: SuiteResult[] = [];
+    let testResult: SolidityTestResult | undefined = undefined;
+
+    if (generate_gas_report) {
+      [testResult, suiteResults] = await runAllSolidityTestsWithGasReport(
+        this.edrContext,
+        chainType,
+        this.artifacts,
+        testContracts,
+        this.tracingConfig,
+        {
+          ...this.defaultConfig(chainType),
+          ...config,
+          generateGasReport: true,
+        }
+      );
+    } else {
+      suiteResults = await runAllSolidityTests(
+        this.edrContext,
+        chainType,
+        this.artifacts,
+        testContracts,
+        this.tracingConfig,
+        {
+          ...this.defaultConfig(chainType),
+          ...config,
+        }
+      );
```

Comment:
Suggestion: As mentioned above, it feels like this duplication can be avoided by modifying the original function to return gas reports.